hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -154,6 +155,12 @@ public class HoodieTableConfig extends HoodieConfig {
       .withDocumentation("Payload class to use for performing compactions, i.e merge delta logs with current base file and then "
           + " produce a new base file.");
 
+  public static final ConfigProperty<String> COMBINE_ENGINE_CLASS_NAME = ConfigProperty
+      .key("hoodie.compaction.combine.engine.class")
+      .defaultValue(HoodieAvroRecordCombiningEngine.class.getName())
```

Comment:
so the same table be read by Spark or Hive for e.g I would love for such queries to be able to read and merge using the engine' POJO (Spark -> Row, Hive -> ArrayWritable), if the user chooses to write such a "combining engine" implementation. My thought here is that we should be able to discover whether such ability exists for class here.

Reply:
@vinothchandar to confirm, you meant when users choose to query with hive for e.g., the config can detect it and choose hive combining engine automatically instead of the default avro-based one?

@wulei0302 @wzx140 we can file ticket to follow up on implementing this detection.

Reply:
https://issues.apache.org/jira/browse/HUDI-4301

Reply:
I updated the ticket. I feel `HoodieMerge` by itself should be designed as engine aware somehow